### PR TITLE
User-provided environment variables

### PR DESCRIPTION
**Pull request by xsc**
See #22.

It might seem a bit weird at first glance, but I think that having a file `.buildenv` that describes which variables to import from the environment is preferable to a lot of `-e ENV=VALUE -e ENV=VALUE ...` calls.

So, if you need `LEIN_PASSWORD`, you'd create a `.buildenv` with:

```txt
LEIN_PASSWORD
```

Additionally, this won't require adapting the Github Action, plus we're able to fail-fast when required environment variables are missing.

### CODE REVIEW

**Comment by holyjak** [comment]
Hi, thanks a lot! The proposal looks fine to me.

File: doc/WHITEPAPER.md [hunk 1]
```diff
@@ -302,6 +302,26 @@ rm -f /usr/share/nginx/html/index.html
...
+leads to undocumented functionality - which should be avoided.
+
+Since the codebase - or the tooling it relies on - is the thing in need of those
+environment variables, there should be a way for it to communicate this fact.
+This, in turn, allows for early validation: Don't even start a build if the
+respective values are missing.
+
+This is achieved by expecting a file `.buildenv` in the source directory that is
+being built. This file will contain one line with each environment variable that
+needs to be provided to ensure a successful build.
```
Comment (holyjak) [inline doc/WHITEPAPER.md]: I would clarify that

> Each variable listed in the file is read from the environment and provided to the build process.

I would also clarify "expecting". Is it obligatory or optional? If the latter then perhaps rephrase to

> This is achieved by allowing a file `.buildenv` in the source directory that is
being built.

Comment (xsc) [reply doc/WHITEPAPER.md]: Good points! I made some adjustment to the description and hope that it's now clearer.

**Review by holyjak — APPROVED** [review]
great, thanks!

**Pull request merged**